app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -284,6 +297,32 @@ class SearchFragment : Fragment(), BackHandler {
         } else null
     }
 
+    /**
+     * Attempts to get a cached search engine. If none is immediately available, it falls back
+     * to a default.
+     *
+     * Cache hits/misses are reported to telemetry.
+     */
+    private fun getCurrentSearchEngine(context: Context): SearchEngine {
+        val deferredSearchEngine = viewLifecycleOwner.lifecycleScope.async {
+            context.components.search.searchEngineManager.getDefaultSearchEngineAsync(context)
+        }
+
+        val selectedSearchEngine = if (deferredSearchEngine.isCompleted) {
+            runBlocking { deferredSearchEngine.await() }
+        } else {
+            null
+        }
+
+        val backupSearchEngine = context.components.search.localeSearchEngineManager
+            .getDefaultSearchEngine(context)
+
+        val cacheWasHit = selectedSearchEngine != null
+        Events.searchFragmentCacheAccess.set(cacheWasHit)
```

Comment:
This is the first time I've added a new ping, would you please double check that I'm doing it properly?